src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp

```diff
@@ -30,1039 +24,938 @@
 
-        SpellScript* GetSpellScript() const override
-        {
-            return new spell_xt002_submerged_SpellScript();
-        }
+    void Register() override
+    {
+        DoCheckProc += AuraCheckProcFn(spell_xt002_321_boombot_aura::CheckProc);
+        OnEffectProc += AuraEffectProcFn(spell_xt002_321_boombot_aura::HandleProc, EFFECT_0, SPELL_AURA_DUMMY);
+    }
 };
 
-class spell_xt002_321_boombot_aura : public SpellScriptLoader
+// 63849 - Exposed Heart
+class spell_xt002_exposed_heart : public AuraScript
 {
-    public:
-        spell_xt002_321_boombot_aura() : SpellScriptLoader("spell_xt002_321_boombot_aura") { }
+    PrepareAuraScript(spell_xt002_exposed_heart);
 
-        class spell_xt002_321_boombot_aura_AuraScript : public AuraScript
-        {
-            PrepareAuraScript(spell_xt002_321_boombot_aura_AuraScript);
-
-            bool Validate(SpellInfo const* /*spellInfo*/) override
-            {
-                return ValidateSpellInfo({ SPELL_ACHIEVEMENT_CREDIT_NERF_SCRAPBOTS });
-            }
+    bool Load() override
+    {
+        _damageAmount = 0;
+        return true;
+    }
 
-            bool CheckProc(ProcEventInfo& eventInfo)
-            {
-                if (eventInfo.GetActionTarget()->GetEntry() != NPC_XS013_SCRAPBOT)
-                    return false;
-                return true;
-            }
+    void OnProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
+    {
+        PreventDefaultAction();
+        DamageInfo* damageInfo = eventInfo.GetDamageInfo();
+        if (!damageInfo || !damageInfo->GetDamage())
+            return;
 
-            void HandleProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
-            {
-                InstanceScript* instance = eventInfo.GetActor()->GetInstanceScript();
-                if (!instance)
-                    return;
+        _damageAmount += damageInfo->GetDamage();
+    }
 
-                instance->DoCastSpellOnPlayers(SPELL_ACHIEVEMENT_CREDIT_NERF_SCRAPBOTS);
-            }
+    void HandleLifeTransfer(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (InstanceScript* instance = GetTarget()->GetInstanceScript())
+            if (Creature* xt002 = instance->GetCreature(BOSS_XT002))
+                xt002->AI()->SetData(DATA_TRANSFERED_HEALTH, _damageAmount);
+    }
 
-            void Register() override
-            {
-                DoCheckProc += AuraCheckProcFn(spell_xt002_321_boombot_aura_AuraScript::CheckProc);
-                OnEffectProc += AuraEffectProcFn(spell_xt002_321_boombot_aura_AuraScript::HandleProc, EFFECT_0, SPELL_AURA_DUMMY);
-            }
-        };
+    void Register() override
+    {
+        OnEffectProc += AuraEffectProcFn(spell_xt002_exposed_heart::OnProc, EFFECT_0, SPELL_AURA_MOD_DAMAGE_PERCENT_TAKEN);
+        AfterEffectRemove += AuraEffectRemoveFn(spell_xt002_exposed_heart::HandleLifeTransfer, EFFECT_0, SPELL_AURA_MOD_DAMAGE_PERCENT_TAKEN, AURA_EFFECT_HANDLE_REAL);
```

Comment:
Don't think the health is transferred when the aura is removed?

If you bring the boss to 5% and let his heart pop out, isn't he going to die way before the heart reaches 0?

Reply:
Is impossible heart pop out in 5% of life in 3.3.5
And i'm pretty sure that life is transfered when heart back to body.
i will try confirm it in retail today.

Reply:
Doesn't matter if it's 3.3.5 or not?

Takes a while for the boss to drop it, although it only triggers on 75/50/25...
Just need to push it from 25% to 5%, wait for heart, and then damage the heart.

Boss should die before the heart reaches 0.

Don't we have sniffs that can confirm this? :P

Reply:
Nope :/
I cant reproduce it in retail, my characters cant remove a good portion of life
Or remove too much or just kill him lul